### PR TITLE
(PC-12471) fix(HonorStatement): refetch /me properly and invalidate /subsciption/next_step

### PR DESCRIPTION
**Pull request by anoukhello**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-12471

## Checklist

I have:

- [x] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** or **github dev name** for any added TODO / FIXME.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)
- [ ] Made sure translations file is up to date (`yarn translations:extract`)

## Screenshots

| iOS | Android | Mobile - Chrome | Desktop - Chrome | Desktop - Safari |
| --: | ------: | --------------: | ---------------: | ---------------: |
|     |         |                 |                  |                  |

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/features/identityCheck/pages/confirmation/IdentityCheckHonor.tsx [hunk 1]
```diff
@@ -23,13 +23,13 @@ export const IdentityCheckHonor = () => {
...
 
   const { mutate: postHonorStatement, isLoading } = usePostHonorStatement({
     onSuccess: () => {
-      queryClient.invalidateQueries(QueryKeys.USER_PROFILE)
-      api
-        .getnativev1me()
-        .then((userProfile) => {
+      queryClient.invalidateQueries(QueryKeys.NEXT_SUBSCRIPTION_STEP)
+      refetch()
+        .then(({ data: userProfile }) => {
```
Comment (antoinewg) [inline src/features/identityCheck/pages/confirmation/IdentityCheckHonor.tsx:32]: 👍  :)

**Review by antoinewg — APPROVED** [review]
ça corrige un bug ?

**Review by Lucasbeneston — APPROVED** [review]
Top ! 👍

File: src/features/identityCheck/pages/confirmation/IdentityCheckHonor.tsx (hunk 1 above)
Comment (anoukhello) [reply src/features/identityCheck/pages/confirmation/IdentityCheckHonor.tsx:32]: Yes au retour dans le profile on avait encore le bouton "Profite de 300e" parce qu'on mettait pas à jour le cache de `/subscription/next_step`

Comment (antoinewg) [reply src/features/identityCheck/pages/confirmation/IdentityCheckHonor.tsx:32]: ah ben https://github.com/pass-culture/pass-culture-app-native/pull/2314#discussion_r771576673 héhé

**Pull request merged**